src/DIRAC/Core/Tornado/Client/ClientSelector.py

```diff
@@ -66,28 +66,30 @@ def ClientSelector(disetClient, *args, **kwargs):  # We use same interface as RP
         # If we are not already given a URL, resolve it
         if serviceName.startswith(("http", "dip")):
             completeUrl = serviceName
-        elif useLegacyAdapter(serviceName):
-            sLog.debug(f"Using legacy adapter for service {serviceName}")
-            if diracxClient is None:
-                raise NotImplementedError(
-                    "DiracX is enabled but no diracxClient is provided, do you need to update your client?"
-                )
-            return diracxClient()
         else:
+            isServiceLegacyAdapted = useLegacyAdapter(serviceName)
+            if isServiceLegacyAdapted and diracxClient is not None:
+                sLog.debug(f"Using legacy adapter for service {serviceName}")
+                # Maybe it comes from the fact that diracxclient is not None for whatever reason
+                sLog.debug(diracxClient)
+                return diracxClient()
+            else:
+                if isServiceLegacyAdapted:
+                    # Only warn to deactivate it without making the code crash.
+                    sLog.warn("FutureClient is provided but not activated, do you need to update your client?")
+
             completeUrl = getServiceURL(serviceName)
             sLog.debug(f"URL resolved: {completeUrl}")
 
         if completeUrl.startswith("http"):
             sLog.debug(f"Using HTTPS for service {serviceName}")
-            rpc = tornadoClient(*args, **kwargs)
-        else:
-            rpc = disetClient(*args, **kwargs)
+            return tornadoClient(*args, **kwargs)
     except Exception as e:  # pylint: disable=broad-except
         # If anything went wrong in the resolution, we return default RPCClient
         # So the behaviour is exactly the same as before implementation of Tornado
         sLog.warn("Could not select DISET or Tornado client", f"{repr(e)}")
-        rpc = disetClient(*args, **kwargs)
-    return rpc
+        print(traceback.format_exc())
```

Comment:
You do not need this, instead call
```suggestion
        sLog.exception()
```